src/user.ts

```diff
@@ -395,6 +399,28 @@ export class User {
             }
         }
     }
+
+    async signMessage(payload: any, recipient?: string): Promise<Message> {
+        const signer = createVCSigner(this.keyManager, KeyManagerLevel.LOCAL);
+
+        const issuer: Issuer = {
+            did: (await this.getDid()) + '#local',
+            signer: signer.sign as any,
+            alg: 'ES256K-R',
+        };
+
+        return await Message.sign(payload, issuer, recipient);
+    }
+
+    async getDid() {
+        if (!this.chainID) {
```

Comment:
instead of caching the `chainID` better if we cache the DID. Actually, this would be better to add to `storage.did` on create account or login. New Task